extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentKiller.java

```diff
@@ -99,13 +118,20 @@ public void killAll() throws IOException
     fs.delete(storageDirectory, true);
   }
 
-  private boolean safeNonRecursiveDelete(FileSystem fs, Path path)
+  private void deleteEmptyParents(FileSystem fs, Path path)
   {
+    if(path.equals(storageDirectory)) {
+      return;
+    }
     try {
-      return fs.delete(path, false);
+      if(fs.listStatus(path).length == 0)
```

Comment:
can this be `if (fs.listStatus(path).length == 0 && fs.delete(path,false)) ` so that delete call return status is not ignored